models/clm/CAM_bias_correction/CAM4_bias_correct.m

```diff
@@ -0,0 +1,1076 @@
+        ncwriteatt(ncname,'doma_area','standard_name','cell area')
+        ncwriteatt(ncname,'doma_area','internal_dname','doma')
+        
+        nccreate(ncname,'doma_lat','Dimensions',{'doma_nx',1,'doma_ny',1},'Format','classic');
+        ncwrite(ncname,'doma_lat',SITE_lat)
+        ncwriteatt(ncname,'doma_lat','_FillValue', 0.)
+        ncwriteatt(ncname,'doma_lat','units','degrees north')
+        ncwriteatt(ncname,'doma_lat','long_name','undefined')
+        ncwriteatt(ncname,'doma_lat','standard_name','latitude')
+        ncwriteatt(ncname,'doma_lat','internal_dname','doma')
+        
+        nccreate(ncname,'doma_lon','Dimensions',{'doma_nx',1,'doma_ny',1},'Format','classic');
+        ncwrite(ncname,'doma_lon',SITE_lon)
+        ncwriteatt(ncname,'doma_lon','_FillValue', 0.)
+        ncwriteatt(ncname,'doma_lon','units','degrees east')
+        ncwriteatt(ncname,'doma_lon','long_name','undefined')
+        ncwriteatt(ncname,'doma_lon','standard_name','longitude')
+        ncwriteatt(ncname,'doma_lon','internal_dname','doma')
+
+        nccreate(ncname,'doma_mask','Dimensions',{'doma_nx',1,'doma_ny',1},'Format','classic');
+        ncwrite(ncname,'doma_mask',SITE_doma_mask)
+        ncwriteatt(ncname,'doma_mask','_FillValue', 0.)
+        ncwriteatt(ncname,'doma_mask','units','unitless')
+        ncwriteatt(ncname,'doma_mask','long_name','undefined')
+        ncwriteatt(ncname,'doma_mask','standard_name','mask')
+        ncwriteatt(ncname,'doma_mask','internal_dname','doma')
+        
+        nccreate(ncname,'a2x6h_Faxa_swndf','Dimensions',{'a2x6h_nx',1,'a2x6h_ny',1,'time',Inf},'Datatype','single','Format','classic');
+        ncwrite(ncname,'a2x6h_Faxa_swndf',swndf_assign)
+        ncwriteatt(ncname,'a2x6h_Faxa_swndf','_FillValue', single(1.e30))
+        ncwriteatt(ncname,'a2x6h_Faxa_swndf','units','W m-2')
+        ncwriteatt(ncname,'a2x6h_Faxa_swndf','long_name','Diffuse near-infrared incident solar radiation')
+        ncwriteatt(ncname,'a2x6h_Faxa_swndf','standard_name','surface_downward_diffuse_shortwave_flux_due_to_near_infrared_radiation')
+        ncwriteatt(ncname,'a2x6h_Faxa_swndf','internal_dname','a2x6h')
+        ncwriteatt(ncname,'a2x6h_Faxa_swndf','cell_methods', 'time: mean')
+
+        nccreate(ncname,'a2x6h_Faxa_swndr','Dimensions',{'a2x6h_nx',1,'a2x6h_ny',1,'time',Inf},'Datatype','single','Format','classic');
+        ncwrite(ncname,'a2x6h_Faxa_swndr',swndr_assign)
+        ncwriteatt(ncname,'a2x6h_Faxa_swndr','_FillValue', single(1.e30))
+        ncwriteatt(ncname,'a2x6h_Faxa_swndr','units','W m-2')
+        ncwriteatt(ncname,'a2x6h_Faxa_swndr','long_name','Direct near-infrared incident solar radiation')
+        ncwriteatt(ncname,'a2x6h_Faxa_swndr','standard_name','surface_downward_direct_shortwave_flux_due_to_near_infrared_radiation')
+        ncwriteatt(ncname,'a2x6h_Faxa_swndr','internal_dname','a2x6h')
+        ncwriteatt(ncname,'a2x6h_Faxa_swndr','cell_methods', 'time: mean') 

```

Comment:
These blocks to write each variable are quite repetitive. You could create a function to create the variable and write its attributes.  That would make your life easier, for example, if you wanted to change the netcdf format to 64-bit Offset you would only need to change the 'format' in one place.

Reply:
Thanks @kdraeder and @hkershaw-brown, I added several commits based on your feedback, reducing repetition by generating `load_towermet.m`, `load_CAM.m`, `write_netcdf_att.m` and `plot_diagnostic` functions.   Same functionality as before.  A step in the right direction.